ecs-agent/tmds/handlers/v4/state/response.go

```diff
@@ -31,12 +32,13 @@ const (
 // with the v2 task response object.
 type TaskResponse struct {
 	*v2.TaskResponse
-	Containers              []ContainerResponse      `json:"Containers,omitempty"`
-	VPCID                   string                   `json:"VPCID,omitempty"`
-	ServiceName             string                   `json:"ServiceName,omitempty"`
-	ClockDrift              *ClockDrift              `json:"ClockDrift,omitempty"`
-	EphemeralStorageMetrics *EphemeralStorageMetrics `json:"EphemeralStorageMetrics,omitempty"`
-	CredentialsID           string                   `json:"-"`
+	Containers              []ContainerResponse                  `json:"Containers,omitempty"`
+	VPCID                   string                               `json:"VPCID,omitempty"`
+	ServiceName             string                               `json:"ServiceName,omitempty"`
+	ClockDrift              *ClockDrift                          `json:"ClockDrift,omitempty"`
+	EphemeralStorageMetrics *EphemeralStorageMetrics             `json:"EphemeralStorageMetrics,omitempty"`
+	CredentialsID           string                               `json:"-"`
+	TaskNetworkConfig       *tasknetworkconfig.TaskNetworkConfig `json:"-"`
```

Comment:
Q: Should we also have a boolean in the task response on whether a task is fault-enabled?

Reply:
Yes, we should. I'm thinking to add this after the HLD is finalized.